services/httpd/handler.go

```diff
@@ -154,6 +156,19 @@ func NewHandler(c Config) *Handler {
 		writeLogEnabled = false
 	}
 
+	var authWrapper func(handler func(http.ResponseWriter, *http.Request)) interface{}
+	if h.Config.AuthEnabled && h.Config.PingAuthEnabled {
+		authWrapper = func(handler func(http.ResponseWriter, *http.Request)) interface{} {
+			return func(w http.ResponseWriter, r *http.Request, user meta.User) {
```

Comment:
I'm looking for something about auth here, but this just looks like a no-op handler wrapper. Can you explain?

Reply:
[`AddRoutes`](https://github.com/influxdata/influxdb/pull/15222/files#diff-719dcaf997cf7e555e32df7f85e7cd60R390) will wrap the handler in `authenticate` if it matches the `func(w http.ResponseWriter, r *http.Request, user meta.User)` signature.

Reply:
[`AddRoutes`](https://github.com/influxdata/influxdb/pull/15222/files#diff-719dcaf997cf7e555e32df7f85e7cd60R390) will wrap the handler in `authenticate` if it matches the `func(w http.ResponseWriter, r *http.Request, user meta.User)` signature.